src/measure_theory/function/ess_sup.lean

```diff
@@ -139,6 +139,37 @@ filter.eventually_lt_of_limsup_lt hf
 lemma ae_lt_of_lt_ess_inf {f : α → β} {x : β} (hf : x < ess_inf f μ) : ∀ᵐ y ∂μ, x < f y :=
 @ae_lt_of_ess_sup_lt α (order_dual β) _ _ _ _ _ hf
 
+lemma ess_sup_indicator_eq_ess_sup_restrict [has_zero β] {s : set α}
+  {f : α → β} (hf : 0 ≤ᵐ[μ.restrict s] f) (hs : measurable_set s) (hs_not_null : μ s ≠ 0) :
+  ess_sup (s.indicator f) μ = ess_sup f (μ.restrict s) :=
+begin
+  simp_rw ess_sup,
+  refine le_antisymm _ _,
+  swap, { exact Limsup_le_Limsup_of_le (map_restrict_ae_le_map_indicator_ae hs)
+    (by is_bounded_default) (by is_bounded_default), },
```

Comment:
To avoid the swap, you can fill in directly the second goal, as in
```suggestion
  refine le_antisymm _ (Limsup_le_Limsup_of_le (map_restrict_ae_le_map_indicator_ae hs)
    (by is_bounded_default) (by is_bounded_default)),
```